go/builder.go

```diff
@@ -540,6 +543,24 @@ func (b *Builder) Slot(slotnum int) {
 	b.vtable[slotnum] = UOffsetT(b.Offset())
 }
 
+// FinishWithFileIdentifier finalizes a buffer, pointing to the given `rootTable`.
+// as well as applys a file identifier
+func (b *Builder) FinishWithFileIdentifier(rootTable UOffsetT, fid []byte) {
+	if fid != nil && len(fid) == fileIdentifierLength {
```

Comment:
rather than silently failing if the length is incorrect, this should probably be a panic?